### DISCUSSION TITLE
Mapping for table and column names

### DISCUSSION THREAD

**Discussion by olmohake** [nextauthjs/next-auth#11560] [Ideas]
### Goals

1. Enable mapping to custom table and column names
2. Enable custom id types

### Background

Authjs is very opinionated about table and column names and types.
This is somewhat problematic when trying to get it up an running against an existing database. It would be very helpful to enable a mapping of default to custom table and column names and support other id field types

### Proposal

```json
{
   adapter:{},
   providers:[],
   tables:{
        users:{
            name: user,
            fields: {
                
            }
       }
   }
}
```

**Comment by skeddles** [comment]
I'd love to be able to host multiple apps on the same database but currently cannot as they would all try to use the same tables.
